### ISSUE TITLE
plugin broken - either the last update, or HTTPS stuff

### ISSUE THREAD

**Issue by metallikat36** [dsifford/academic-bloggers-toolkit#291]
**ABT Version:** 4.10.0
**PHP Version:** 5.6.29
**Theme:** Dazzling
Browser: Out of date Chrome on a Vista machine

Plugin List:
Academic Blogger's Toolkit
Ajax Search Lite
All In One WP Security
CM Table Of Contents Pro
Disqus Conditional Load
Easy Google Fonts
Give - Donation Plugin
Hyper Cache
Jetpack
Popular Posts Tabbed Widget for Jetpack
PopupAlly
Post Slider WD
Scalable Vector Graphics (SVG)
Social Popscroll
Super Socializer
UpdraftPlus - Backup/Restore
WP Smush
Yoast SEO

**Issue:**

Hi Derek,

I really hope this one doesn't open up a can of worms. I am in the process of moving my site from HTTP to HTTPS (i.e. uses SSL certificates, gets that little green padlock in the url bar that tells users the site is secure, allows you to do credit card transactions safely on your website, etc). I started doing this stuff today. And I know you also rolled out a plugin update yesterday. So I don't know which of these two is the cause of the following bug for sure right now. If you want me to test, and can guide me as to how to rollback the plugin, I can test and see if an older plugin solves the issue.

Basically the issue is what you see below. If you try to add a reference, it just sits forever and the text field never appears for you to enter a PMID:
![image](https://cloud.githubusercontent.com/assets/17968340/22180854/11cf0894-e032-11e6-95b0-20e9f7e36b52.png)

This page has a little bit of info about possible issues with JS and switching to HTTPS.
https://givewp.com/documentation/resources/how-to-use-ssl-and-https-for-your-wordpress-website/
I also read elsewhere that issues with plugins might be fixed with "“protocol relative URLs." Hopefully within the next day or two, I will have done some tests to determine if something about ABT is considered "insecure" or not. 

David

**Comment by dsifford** [comment]
I use the plugin on a few different sites that all have https enabled, so that's definitely not the issue.

Happy to help troubleshoot though if you haven't already figured it out.

Re: protocol URLs... Every external url used in the plugin is also SSL encrypted, so you should have no issue there.

**Comment by metallikat36** [comment]
I figured out the issue. In the course of setting my my SSL certificate with Cloudflare, I also enabled a feature they have called Rocket Loader. When this feature is enabled, I get the bug I described. When it's disabled, problem solved.

I also would get two others bugs when it is enabled When I would open a post in a browser for the first time, the tooltips would not work, and the references list would be in the expanded state and would not collapse. Then if I would reload the page (or navigate to a second post) then everything would work fine. It's like it was not loading the entire code in one pass but needed two passes to do it. 

So the fix for me for now is to just not enable Rocket Loader. In case you aren't familiar with it, here is a screenshot from my Cloudflare Dashboard:
![image](https://cloud.githubusercontent.com/assets/17968340/22192401/aafb88b0-e0e8-11e6-9467-e709145f544e.png)

The only thing else I know about Rocket Loader is that it used to not work well with DISQUS comments, but they have since updated to plugin to make it work.

**Comment by dsifford** [comment]
Not totally sure what could be causing the specific issue, but a few things that I do know are:
- I bundle my code in a very specific format and in a very specific order which can't be changed for it to work properly.
- The scripts that are loaded at the bottom of the `<body>` don't need to be asynchronous since that location doesn't block anything. 
- The reason why it doesn't work on the first time, but does on the second time is because it sounds like your browser is caching it.

Side note: Hope you didn't pay any money for those SSL certificates. You can get them for free now with Let's Encrypt.

**Comment by metallikat36** [comment]
Okay, thanks. So I will just not use Rocket Loader with ABT, then. No problem. 

No I did not pay for SSL. Siteground gives free Let's Encrypt SSL and Cloudflare some other free shared SSL.

**Comment by dsifford** [comment]
@metallikat36 You'll get a better performance benefit if you enable the HTTP/2 protocol vs using Rocket Loader. 

The difference is....
- **Rocket Loader** combines scripts into a single giant network request which solves the issue of having a single thread of network requests in a series for all of your resources one-by-one.
- **HTTP/2** loves multiple network requests. HTTP/2 operates on a single TCP connection in both directions and multiplexes the resource downloads so that they all run in parallel. HTTP/2-enabled sites are also given a higher SEO score by google vs HTTP/1.1 with all other things equal.

**Bottom Line:** HTTP/2 is a better use of your time than Rocket Loader.
